atlassian/connector.go

```diff
@@ -0,0 +1,85 @@
+package atlassian
+
+import (
+	"context"
+	"fmt"
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/interpreter"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+	"github.com/amp-labs/connectors/providers"
+)
+
+type Connector struct {
+	BaseURL   string
+	Module    string
+	Workspace string
+	Client    *common.JSONHTTPClient
+}
+
+func NewConnector(ctx context.Context, opts ...Option) (conn *Connector, outErr error) {
+	defer common.PanicRecovery(func(cause error) {
+		outErr = cause
+		conn = nil
+	})
+
+	params, err := parameters{}.FromOptions(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	httpClient := params.Client.Caller
+	conn = &Connector{
+		Client: &common.JSONHTTPClient{
+			HTTPClient: httpClient,
+		},
+		Workspace: params.Workspace.Name,
+	}
+
+	providerInfo, err := providers.ReadInfo(conn.Provider())
+	if err != nil {
+		return nil, err
+	}
+
+	// connector and its client must mirror base url and provide its own error parser
+	conn.setBaseURL(providerInfo.BaseURL)
+	conn.Client.HTTPClient.ErrorHandler = interpreter.ErrorHandler{
+		JSON: conn.interpretJSONError,
+	}.Handle
+
+	// CloudID is dynamic data and cannot be supplied via ProviderInfo.
+	// This information will be filled by making discovery API calls.
+	cloudID, err := conn.getCloudID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	conn.Module, err = params.Module.ResolveSuffix(map[string]string{
```

Comment:
Module is a template awaiting for cloud id variable.
Connector hold this state to construct URLs. See methods below in this file.